watch/attr_controllers.py

```diff
@@ -15,14 +15,16 @@ def __getattr__(self, attr_name):
 
     def __get__(self, obj, klass=None):
         # when attr being looked up in class instead of instance
-        if klass is not None and obj is None:
+        # klass is always not None
+        if obj is None:
             return self
 
-        if self.field_name not in obj.__dict__:
```

Comment:
Though this patch will make ```watch``` run slower on attribute error scenario, it is still considered to be a step forward performance-wise.